lib/core/src/Cardano/Wallet.hs

```diff
@@ -1613,6 +1609,57 @@ forgetTx ctx wid tid = db & \DBLayer{..} -> do
   where
     db = ctx ^. dbLayer @s @k
 
+-- | Calculate the transaction expiry slot, given a 'TimeInterpreter', and an
+-- optional TTL in seconds.
+--
+-- If no TTL is provided, a default of 2 hours is used (note: there is no
+-- particular reason why we chose that duration).
+getTxExpiry
+    :: TimeInterpreter (ExceptT PastHorizonException IO)
+    -- ^ Context for time to slot calculation.
+    -> Maybe NominalDiffTime
+    -- ^ Time to live (TTL) in seconds from now.
+    -> IO SlotNo
+getTxExpiry ti maybeTTL = do
+    expTime <- addRelTime ttl <$> currentRelativeTime (unsafeExtendSafeZone ti)
+    interpretQuery (unsafeExtendSafeZone ti) $ ceilingSlotAt expTime
+  where
+    ttl = fromMaybe defaultTTL maybeTTL
+
+    defaultTTL :: NominalDiffTime
+    defaultTTL = 7200  -- that's 2 hours
+
+-- | Retry submission of pending transactions.
+runLocalTxSubmissionPool
+    :: forall ctx s k a.
+        ( HasLogger WalletLog ctx
+        , HasNetworkLayer ctx
+        , HasDBLayer s k ctx
+        )
+    => WalletId
+    -> ctx
+    -> IO a
+runLocalTxSubmissionPool wid ctx = db & \DBLayer{..} -> forever
+    watchNodeTip $ \bh -> bracketTracer trBracket $ do
+        -- TODO: ADP-764 limit pool check frequency to e.g. every 1000ms at most
+        pending <- atomically $ getLocalTxSubmissionPending (PrimaryKey wid)
+        -- re-submit those transactions, ignore errors
+        forM_ pending $ \st -> do
+            -- TODO: compare slots to schedule retries less often
```

Comment:
Ah! That's exactly the bit I was looking for :no_mouth:; no schedule _yet_ that is.

Reply:
We probably won't need anything more sophisticated that resend every _n_ slots.
(Maybe resending every slot is overkill, but perhaps not).